ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -0,0 +1,201 @@
+
+}
+
+# Check version, state, verified in history
+# Parameter: 1- release version
+function check_history() {
+    local history; history=$(./oc get clusterversion/version -o json)
+    local version; version=$(echo "${history}" | jq -r '.status.history[0].version')
+    local state; state=$(echo "${history}" | jq -r '.status.history[0].state')
+    local verified; verified=$(echo "${history}" | jq -r '.status.history[0].verified')
+    if [[ "${version}" == "$1" && "${state}" == "Completed" ]]; then
+        echo "History check PASSED, cluster is now upgraded to $1"
+    else
+        echo >&2 "History check FAILED, cluster upgrade to $1 failed, current version is ${version}, exiting" && exit 1
+    fi
+    if [[ "${FORCE_UPDATE}" == "false" ]]; then
+        if [[ "${signed}" == "True" && "${verified}" == "true" || "${signed}" == "False" && "${verified}" == "false" ]]; then
+            echo "Verified state check PASSED"
+        else
+            echo >&2 "Verified state check FAILED, exiting" && exit 1
+        fi
+    else
+        if [[ "${verified}" == "false" ]]; then
+            echo "Verified state check PASSED"
+        else
+            echo >&2 "Verified state check FAILED, exiting" && exit 1
+        fi
+    fi
+}
+
+# Setup proxy if it's present in the shared dir
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1091
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+# Note: RELEASE_IMAGE_xxx is by-digest pull spec
+# oc registry login creates registry credentials and then the oc adm release info command works 
+oc registry login
+
+get_target_release
+for target in "${TARGET_RELEASES[@]}"
+do       
+    source_major_version="$(oc get clusterversion --no-headers | awk '{print $2}' | cut -f1 -d.)"
+    source_minor_version="$(oc get clusterversion --no-headers | awk '{print $2}' | cut -f2 -d.)"
+    echo -e "Source release major version is: ${source_major_version}\nSource release minor version is: ${source_minor_version}"
+       
+    target_registry="$(echo "${target}" | cut -f1 -d@)"
+    target_version="$(oc adm release info "${target}" --output=json | jq -r '.metadata.version')"
+    target_major_version="$(echo "${target_version}" | cut -f1 -d.)"
+    target_minor_version="$(echo "${target_version}" | cut -f2 -d.)"
+    target_tag="${target_version}"
+    [[ ${target_version} != *"nightly"* ]] && target_tag="${target_version}-x86_64"
+    echo -e "Target release is: ${target}\nTarget release version is: ${target_version}\nTarget release registry is: ${target_registry}\nTarget major version is: ${target_major_version}\nTarget minor version is: ${target_major_version}\nTarget tag is: ${target_tag}"
+    
+    extract_oc "${target}"
+
+    signed="$(validate_signed_image "${target}")"
+
+    if [[ "${FORCE_UPDATE}" == "false" ]]; then
+        admin_ack "${source_minor_version}" "${target_minor_version}"
+    fi
+    
+    upgrade "${target}" "${target_registry}:${target_tag}"
```

Comment:
QE are suggested and we are encourage customers to run upgrade using sha256 digest image as target image instead of tag name when running upgrade, so here we should follow that to use digest pullspec as target image.